pkg/providers/vault/secret_manager_test.go

```diff
@@ -20,7 +20,10 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
+	"reflect"
 	"testing"
+
+	errors2 "github.com/edgexfoundry/go-mod-secrets/pkg/errors"
```

Comment:
To get rid of this, consider moving `go-mod-secrets/pkg/errors/types.go` to `go-mod-secrets/pkg/errors.go`

Reply:
Sadly, doing this creates an import cycle. The `client_test.go` file in the `pkg` package imports the `mock` package to use the mock secret manager in the `secret_manager.go` file. The `secret_manager.go` file imports the `pkg` package to use the error `ErrSecretsNotFound`. I cannot think of a clean way around this. Any suggestions?

Reply:
@AnthonyMBonafide See what you think of this approach
https://github.com/tsconn23/go-mod-secrets/tree/pr_11_tsc

Reply:
The only thing with that approach is that my original intention for the `mockSecretStoreManager` was for consumers of the `go-mod-secrets` module to use it for testing, so moving it to the `client_test.go` file kind of goes against that. However, now thinking about it maybe that should not be the intention of that type.

https://github.com/tsconn23/go-mod-secrets/blob/3aa57d6a86fb9852ce9c1c4b0036668f4a7a2278/pkg/client_test.go#L68-L71

Reply:
Let 'em generate their own mock via `mockery`

Reply:
The programming equivalent of "Get off my lawn"!